### PR TITLE
chore: Refactor demo to use our own UI instead of Showkase one

### LINKED ISSUE #220: Build our own layout + navigation and show the **Showcase** previews with the help of the metadata

### PR DESCRIPTION
**Pull request by soulcramer**
## 📋 Changes description

Rebuild the sample app by showing the Showkase metadata instead of letting it handle it

## 🤔 Context

This would make possible for us to customize themes and add interactives demos

## 📸 Screenshots

image
image
image

## 🗒️ Other info

close #220

### CODE REVIEW

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+                topBar = {
+                    AppBar(
+                        navController = navController,
+                        scrollBehavior = scrollBehavior,
+                        showkaseBrowserScreenMetadata = showkaseBrowserScreenMetadata,
+                    )
+                },
+                content = {
+                    BodyContent(
+//                        modifier = Modifier.fillMaxSize(),
```
Comment (SimonMarquis) [inline sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:95]: Dead code?
```suggestion
```

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalSparkApi::class)
+@Composable
+internal fun AppBar(
+    navController: NavHostController,
+    scrollBehavior: TopAppBarScrollBehavior,
+    showkaseBrowserScreenMetadata: MutableState<ShowkaseBrowserScreenMetadata>,
```
Comment (SimonMarquis) [inline sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:112]: nit: let's use simpler variable name, like `metadata`. It makes the code more readable.

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+                contentPadding = contentPadding,
+                showkaseBrowserScreenMetadata,
+                groupedComponentMap,
+            )
+        },
+    )
+}
+
+private fun startDestination(
+) = CurrentScreen.COMPONENT_GROUPS.name
```
Comment (SimonMarquis) [inline sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:333]: nit: Not sure why we need this function

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+
+@Composable
+internal fun BodyContent(
+    modifier: Modifier = Modifier,
+    contentPadding: PaddingValues,
+    navController: NavHostController,
+    groupedComponentMap: Map<String, List<ShowkaseBrowserComponent>>,
+    showkaseBrowserScreenMetadata: MutableState<ShowkaseBrowserScreenMetadata>,
+) {
+    val startDestination = startDestination()
```
Comment (SimonMarquis) [inline sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:316]: nit: could be inlined

File: sample/src/main/kotlin/com/adevinta/spark/sample/ComponentsInAGroupScreen.kt [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    contentPadding: PaddingValues,
+) {
+    val groupByComponentName =
+        groupedComponentMap[showkaseBrowserScreenMetadata.value.currentGroup]
+            ?.groupBy { it.componentName } ?: return
+    // Use the default style as the preview if its available or take the first style for the component
+    val componentList = groupByComponentName.values.map {
+        it.firstOrNull { it.isDefaultStyle } ?: it.first()
+    }
+    val filteredList =
```
[the remark this answers is not in the document]
Comment (soulcramer) [reply sample/src/main/kotlin/com/adevinta/spark/sample/ComponentsInAGroupScreen.kt:62]: I think (hope) so

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+    }
+}
+
+@Composable
+private fun AppBarTitle(
+    modifier: Modifier,
+    currentRoute: String?,
+    currentGroup: String?,
+    currentComponentName: String?,
+    currentComponentStyleName: String?,
```
[the remark this answers is not in the document]
Comment (MarinaRomanova) [reply sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:194]: I agree with Simon, no need for "current"

File: sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    }
+}
+
+
+@Composable
+public fun ToolbarTitle(
+    string: String,
+    modifier: Modifier,
+) {
+    val lineCount = remember { mutableStateOf(0) }
```
Comment (MarinaRomanova) [inline sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:226]: Why are we interested in line's count?

Comment (soulcramer) [reply sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt:226]: I don't remember precisely but I think it's only used for semantics

**Review by MarinaRomanova — APPROVED** [review]
LGTM

**Pull request merged**